cli/src/main/java/com/walmartlabs/concord/cli/runner/secrets/CliSecretService.java

```diff
@@ -0,0 +1,165 @@
+
+        var local = cliConfig.secrets().local();
+        if (local.enabled()) {
+            var provider = new FileSecretsProvider(workDir, local.dir());
+            providers.add(new SecretsProviderRef("localFile", provider, local.writable()));
+        }
+
+        var remote = cliConfig.secrets().remote();
+        if (remote.enabled()) {
+            var provider = new RemoteSecretsProvider(workDir, remote.baseUrl(), remote.apiKey());
+            providers.add(new SecretsProviderRef("remote", provider, remote.writable()));
+        }
+
+        var vault = cliConfig.secrets().vault();
+        return new CliSecretService(providers, new VaultProvider(vault.dir(), vault.id()), verbosity);
+    }
+
+    public CliSecretService(List<SecretsProviderRef> secretsProviders, VaultProvider vaultProvider, Verbosity verbosity) {
+        this.secretsProviders = requireNonNull(secretsProviders);
+        this.vaultProvider = requireNonNull(vaultProvider);
+        this.verbosity = requireNonNull(verbosity);
+    }
+
+    @Override
+    public SecretService.KeyPair exportKeyAsFile(String orgName, String secretName, String secretPassword) throws Exception {
+        for (var ref : secretsProviders) {
+            var result = ref.provider().exportKeyAsFile(orgName, secretName, secretPassword);
+            if (result.isPresent()) {
+                reportSecretFound(orgName, secretName, ref);
+                return result.get();
+            }
+        }
+        throw new RuntimeException("Secret not found: %s/%s.".formatted(orgName, secretName));
+    }
+
+    @Override
+    public String exportAsString(String orgName, String secretName, String secretPassword) throws Exception {
+        for (var ref : secretsProviders) {
+            var result = ref.provider().exportAsString(orgName, secretName, secretPassword);
+            if (result.isPresent()) {
+                reportSecretFound(orgName, secretName, ref);
+                return result.get();
+            }
+        }
+        throw new RuntimeException("Secret not found: %s/%s.".formatted(orgName, secretName));
+    }
+
+    @Override
+    public Path exportAsFile(String orgName, String secretName, String secretPassword) throws Exception {
+        for (var ref : secretsProviders) {
+            var result = ref.provider().exportAsFile(orgName, secretName, secretPassword);
+            if (result.isPresent()) {
+                reportSecretFound(orgName, secretName, ref);
+                return result.get();
+            }
+        }
+        throw new RuntimeException("Secret not found: %s/%s.".formatted(orgName, secretName));
+    }
+
+    @Override
+    public UsernamePassword exportCredentials(String orgName, String secretName, String secretPassword) throws Exception {
+        for (var ref : secretsProviders) {
+            var result = ref.provider().exportCredentials(orgName, secretName, secretPassword);
+            if (result.isPresent()) {
```

Comment:
Giving this a shot trying to look up a remote secret (username/password). Config loads fine, but the first attempt through this loop (with two providers available) ends with an exception so the second attempt (remote) never occurs.

Disabling `secrets.local` fixes that since it only needs to try the one, correct, provider. But it appears this logic is supposed to "find the first successfully loadable secret".

Reply:
Let me re-check this. It was supposed to try each provider until the secret is found.

Reply:
Should be fixed now.